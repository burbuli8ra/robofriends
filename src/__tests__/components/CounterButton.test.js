import React from 'react';
import { shallow } from 'enzyme';
import CounterButton from '../../components/CounterButton';

let wrapper, mockColor;

beforeEach(() => {
  mockColor = 'blue';
  wrapper = shallow(<CounterButton color={ mockColor } />);
});

describe('CounterButton Component', () => {
  it('should render CounterButton component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update on click', () => {
    expect(wrapper.state('count')).toEqual(0);

    wrapper.find('[data-test="counter"]').simulate('click');
    expect(wrapper.state('count')).toEqual(1);

    wrapper.find('[data-test="counter"]').simulate('click');
    expect(wrapper.state('count')).toEqual(2);

    wrapper.find('[data-test="counter"]').simulate('keypress');
    expect(wrapper.state('count')).toEqual(2);

    expect(wrapper.props().style.color).toEqual('blue');
  });

  it('should not update counter', () => {
    const mockNextProps = {
      color: 'blue'
    };

    const mockNextState = {
      count: 0
    };

    expect(wrapper.instance().shouldComponentUpdate(mockNextProps, mockNextState)).toBeFalsy;
  });
});
