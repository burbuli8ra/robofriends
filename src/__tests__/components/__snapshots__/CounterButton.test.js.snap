// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton Component should render CounterButton component 1`] = `ShallowWrapper {}`;
